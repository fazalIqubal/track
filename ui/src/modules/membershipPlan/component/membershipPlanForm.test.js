import React from 'react';
import ReactDOM from 'react-dom';
import {configure, shallow } from 'enzyme';
import { shallowToJson} from 'enzyme-to-json';
import MembershipPlanForm from './membershipPlanForm'
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

const membershipData = {
    name:"Standard",
    description:'6 Month',
    ptice: 299,
    project_limit:8,
    user_limit:6,
    plan_duration:"Six Months"
}
describe('when page is rendering', () => {
    let membershipPlanForm
    const mockDispatch = jest.fn();
    beforeEach(() => {
        membershipPlanForm = shallow(<MembershipPlanForm.WrappedComponent editedMembershipPlanList ={membershipData} dispatch={mockDispatch}/>)
    })
    afterEach(() => {
        membershipPlanForm = null;
    });
    it('should render the form', () => {
        expect(shallowToJson(membershipPlanForm)).toMatchSnapshot()
    });
    it('should render the form lable', () => {
        expect(membershipPlanForm.find('.membershipPlan_name').exists()).toEqual(true);
        expect(membershipPlanForm.find('.description').exists()).toEqual(true);
        expect(membershipPlanForm.find('.price').exists()).toEqual(true);
        expect(membershipPlanForm.find('.project_limit').exists()).toEqual(true);
        expect(membershipPlanForm.find('.user_limit').exists()).toEqual(true);
        expect(membershipPlanForm.find('.plan_duration').exists()).toEqual(true);
    });
    it('should pre fill fields', () => {
        expect(membershipPlanForm.find("#membershipPlan_name").find({name: 'name'}).props().value).toBe(membershipData.name);
        expect(membershipPlanForm.find("#description").find({name: 'description'}).props().value).toBe(membershipData.description);
        expect(membershipPlanForm.find("#price").find({name: 'price'}).props().value).toBe(membershipData.price);
        expect(membershipPlanForm.find("#project_limit").find({name: 'project_limit'}).props().value).toBe(membershipData.project_limit);
        expect(membershipPlanForm.find("#user_limit").find({name: 'user_limit'}).props().value).toBe(membershipData.user_limit);
        expect(membershipPlanForm.find("#plan_duration").find({name: 'plan_duration'}).props().value).toBe(membershipData.plan_duration);
    });
   
});