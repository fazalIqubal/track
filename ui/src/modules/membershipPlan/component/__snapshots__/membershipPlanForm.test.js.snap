// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Membershipplan should renders correctly 1`] = `
<ForwardRef(InternalForm)
  id="membershipPlanForm"
  layout="vertical"
>
  <FormItem
    className="membershipPlan_name"
    label="Membership Plan Name"
    name="name"
    rules={
      Array [
        Object {
          "message": "Membership Plan's Name is required!",
          "required": true,
        },
      ]
    }
  >
    <Input
      id="membershipPlan_name"
      name="name"
      onChange={[Function]}
      type="string"
      value="Standard"
    />
  </FormItem>
  <FormItem
    className="description"
    label="Description"
    name="description"
    rules={
      Array [
        Object {
          "message": "Description is required!",
          "required": true,
        },
      ]
    }
  >
    <Input
      id="description"
      name="description"
      onChange={[Function]}
      type="string"
      value="6 Month"
    />
  </FormItem>
  <FormItem
    className="price"
    label="Price"
    name="price"
    rules={
      Array [
        Object {
          "message": "Price is required",
          "required": true,
        },
      ]
    }
  >
    <Input
      id="price"
      name="price"
      onChange={[Function]}
      type="number"
    />
  </FormItem>
  <FormItem
    className="project_limit"
    label="Project Limit"
    name="project_limit"
    rules={
      Array [
        Object {
          "message": "Project Limit is required",
          "required": true,
        },
      ]
    }
  >
    <Input
      id="project_limit"
      name="project_limit"
      onChange={[Function]}
      type="number"
      value={8}
    />
  </FormItem>
  <FormItem
    className="user_limit"
    label="User Limit"
    name="user_limit"
    rules={
      Array [
        Object {
          "message": "User Limit is required",
          "required": true,
        },
      ]
    }
  >
    <Input
      id="user_limit"
      name="user_limit"
      onChange={[Function]}
      type="number"
      value={6}
    />
  </FormItem>
  <FormItem
    className="plan_duration"
    label="Plan Duration"
    name="plan_duration"
    rules={
      Array [
        Object {
          "message": "Plan Duration is required",
          "required": true,
        },
      ]
    }
  >
    <Select
      bordered={true}
      choiceTransitionName=""
      id="plan_duration"
      name="plan_duration"
      onChange={[Function]}
      placeholder="Select Your Plan"
      transitionName="slide-up"
      value="Six Months"
    >
      <Option
        key="0"
        value="Six Months"
      >
        Six Months
      </Option>
      <Option
        key="1"
        value="One Year"
      >
        One Year
      </Option>
      <Option
        key="2"
        value="Two Years"
      >
        Two Years
      </Option>
      <Option
        key="3"
        value="Three Years"
      >
        Three Years
      </Option>
    </Select>
  </FormItem>
</ForwardRef(InternalForm)>
`;

exports[`when page is rendering should render the form 1`] = `
<ForwardRef(InternalForm)
  id="membershipPlanForm"
  layout="vertical"
>
  <FormItem
    className="membershipPlan_name"
    label="Membership Plan Name"
    name="name"
    rules={
      Array [
        Object {
          "message": "Membership Plan's Name is required!",
          "required": true,
        },
      ]
    }
  >
    <Input
      id="membershipPlan_name"
      name="name"
      onChange={[Function]}
      type="text"
      value="Standard"
    />
  </FormItem>
  <FormItem
    className="description"
    label="Description"
    name="description"
    rules={
      Array [
        Object {
          "message": "Description is required!",
          "required": true,
        },
      ]
    }
  >
    <Input
      id="description"
      name="description"
      onChange={[Function]}
      type="text"
      value="6 Month"
    />
  </FormItem>
  <FormItem
    className="price"
    label="Price"
    name="price"
    rules={
      Array [
        Object {
          "message": "Price is required",
          "required": true,
        },
      ]
    }
  >
    <Input
      id="price"
      name="price"
      onChange={[Function]}
      type="number"
    />
  </FormItem>
  <FormItem
    className="project_limit"
    label="Project Limit"
    name="project_limit"
    rules={
      Array [
        Object {
          "message": "Project Limit is required",
          "required": true,
        },
      ]
    }
  >
    <Input
      id="project_limit"
      name="project_limit"
      onChange={[Function]}
      type="number"
      value={8}
    />
  </FormItem>
  <FormItem
    className="user_limit"
    label="User Limit"
    name="user_limit"
    rules={
      Array [
        Object {
          "message": "User Limit is required",
          "required": true,
        },
      ]
    }
  >
    <Input
      id="user_limit"
      name="user_limit"
      onChange={[Function]}
      type="number"
      value={6}
    />
  </FormItem>
  <FormItem
    className="plan_duration"
    label="Plan Duration"
    name="plan_duration"
    rules={
      Array [
        Object {
          "message": "Plan Duration is required",
          "required": true,
        },
      ]
    }
  >
    <Select
      bordered={true}
      choiceTransitionName=""
      id="plan_duration"
      name="plan_duration"
      onChange={[Function]}
      placeholder="Select Your Plan"
      transitionName="slide-up"
      value="Six Months"
    >
      <Option
        key="0"
        value="Six Months"
      >
        Six Months
      </Option>
      <Option
        key="1"
        value="One Year"
      >
        One Year
      </Option>
      <Option
        key="2"
        value="Two Years"
      >
        Two Years
      </Option>
      <Option
        key="3"
        value="Three Years"
      >
        Three Years
      </Option>
    </Select>
  </FormItem>
</ForwardRef(InternalForm)>
`;
